.coveragerc

```diff
@@ -5,3 +5,4 @@ omit =
     scipy/setup.py
     scipy/*/setup.py
     scipy/signal/_max_len_seq_inner.py
+    **/tests/**
```

Comment:
> FWIW, my view here is that we should explicitly **include** the coverage of the unit tests. This can help to discover cases, for example, where two unit tests have the same name and one of them doesn't get executed, etc.

Ah, yes. Agreed. But it then overreports because... tests are always run (100% coverage) :) Mainly, I want to make it easier to discover uncovered code and check if a pull request doesn't decrease the overall coverage. I don't think that checking tests for coverage would hurt that much so, I can revert this change.

> I think the overall coverage results are always going to be a bit of a mess with current tools, but explicitly ignoring some third-party libraries may be useful I suppose

Sounds good. I will try to add exemptions for third-party code and code generators. I think the coverage reports should be accurate enough after doing that. Then, we can decide if we want to exclude 0% coverage files or not.

Reply:
For this we could just not report these and empty files with the following configuration
```
[tool.coverage.report]
skip_covered = true
skip_empty = true
```